src/selectors/metadata.js

```diff
@@ -0,0 +1,67 @@
+import { createSelector } from 'reselect';
+import { getClickedNode, getNodesById } from './nodes';
+import { runCommandTemplates } from '../config';
+
+/**
+ * Comparison for sorting alphabetically by name, otherwise by value
+ */
+const sortAlpha = (a, b) => (a.name || a).localeCompare(b.name || b);
+
+/**
+ * Returns true if metadata sidebar is visible
+ */
+export const getVisibleMetaSidebar = createSelector(
+  [getClickedNode],
+  nodeClicked => Boolean(nodeClicked)
+);
+
+/**
+ * Returns run command for the node, if applicable to the node type
+ * @param {object} node The node
+ * @returns {?string} The run command for the node, if applicable
+ */
+const getRunCommand = node => {
+  const template = runCommandTemplates[node.type];
+  return template ? template(node.fullName) : null;
+};
+
+/**
+ * Gets metadata for the currently clicked node if any
+ */
+export const getClickedNodeMetaData = createSelector(
+  [
+    getClickedNode,
+    getNodesById,
+    state => state.node.tags,
+    state => state.tag.name,
+    state => state.pipeline
+  ],
+  (nodeId, nodes = {}, nodeTags, tagNames, pipeline) => {
+    const node = nodes[nodeId];
+
+    if (!node) {
+      return null;
+    }
+
+    const metadata = {
+      node,
+      tags: [...nodeTags[node.id]]
+        .map(tagId => tagNames[tagId])
+        .sort(sortAlpha),
+      pipeline: pipeline.name[pipeline.active],
+      runCommand: getRunCommand(node)
+    };
+
+    // Note: node.sources node.targets require newgraph enabled
```

Comment:
That's interesting. It's probably achievable relatively easily with the old Dagre setup, but I imagine it's not worth it as we'll be deprecating that soon anyway

Reply:
Yeah so ideally this logic should be refactored into the selectors, which I started doing to avoid this but there were a lot of complications due to some dependencies and exactly we're due to deprecate so it's not worth that effort at this time, but I'll likely revisit the idea later.